src/main/java/com/autotune/common/datasource/DataSourceManager.java

```diff
@@ -110,6 +110,10 @@ public void deleteMetadataFromDataSource(DataSourceInfo dataSource) {
      */
     public void saveMetadataFromDataSourceToDB(DataSourceInfo dataSourceInfo) {
         try {
+            if (null == dataSourceInfo) {
```

Comment:
@shreyabiradar07 - As discussed, please add the negative scenario to bring down the prometheus by scaling down the statefulset and deployment between multiple imports, and validate that the second call to import metadata fails with appropriate error message and list metadata returns the metadata from the DB. This can be added in a separate PR.

Reply:
@chandrams - created [#1211](https://github.com/kruize/autotune/pull/1211) adding a negative scenario to test multiple metadata imports by bringing down prometheus server instance following the above approach. Please review the same, thanks!